### PR TITLE
Create CI

### PR DESCRIPTION
**Pull request by regulad**
An idea I had for testing commits and pull requests before marking them safe to merge. Most bigger projects implement some sort of CI with artifacts, who why not ninjamuffin9/Funkin?

You might need to change references made to me, like line 70.

### CODE REVIEW

**Comment by regulad** [comment]
Tested, works.

**Review by VoltrexMaster — APPROVED** [review]
👍

**Review by actuallyasmartname — APPROVED** [review]
good

**Comment by Two-Ai** [comment]
nice!
you might want to add Newgrounds and Switch build targets as well (see the build files in `art/build-*.bat`)

**Comment by regulad** [comment]
> 
> 
> nice!
> you might want to add Newgrounds and Switch build targets as well (see the build files in `art/build-*.bat`)

Can't do switch since it requires copyrighted libraries. Newgrounds is just HTML5 with some extra stuff.
A CI is for checking the validity of code, as a side effect providing artifacts.

**Review by Two-Ai — APPROVED** [review]
ah, good point on the Switch stuff
I'd still suggest a Newgrounds build target to make sure the code is valid with and without `ng` defined, since some code is gated off with `#if ng` blocks. That said, it's relatively low impact so it's fine without.

**Comment by aflacc** [comment]
im sorry **switch**???

**Review by hdboye — APPROVED** [review]
is gud

**Comment by aflacc** [comment]
what is cl

**Comment by VoltrexMaster** [comment]
> what is cl

Continuous integration.

**Comment by aflacc** [comment]
> > what is cl
> 
> Continuous integration.

what is that

**Comment by VoltrexMaster** [comment]
> > > what is cl
> > 
> > 
> > Continuous integration.
> 
> what is that

https://en.m.wikipedia.org/wiki/Continuous_integration

**Comment by aflacc** [comment]
> > > > what is cl
> > > 
> > > 
> > > Continuous integration.
> > 
> > 
> > what is that
> 
> https://en.m.wikipedia.org/wiki/Continuous_integration

im too lazy

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,78 @@
+name: CI
+on:
```
Comment (actuallyasmartname) [inline .github/workflows/ci.yml:2]: Probably should set this to all branches

**Review by actuallyasmartname — CHANGES_REQUESTED** [review]

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  push:
+    branches: [ master ]
+  pull_request:
+    branches: [ master ]
+  workflow_dispatch:
```
Comment (actuallyasmartname) [inline .github/workflows/ci.yml:7]: probably should set this to all branches not just master

**Review by actuallyasmartname — CHANGES_REQUESTED** [review]

**Pull request closed without merging**